features/appmgt/org.wso2.carbon.appmgt.publisher.feature/src/main/resources/publisher/apis/v1/validations_api_router.jag

```diff
@@ -1,96 +1,143 @@
+    var router = require('/modules/router-g.js').router();
+    var routeManager = new router.Router();
+    var log = new Log();
+
+    routeManager.register('GET', 'publisher', '/publisher/api/validate/{appType}/field/{fieldName}', function (context) {
+
+        var appType = context.params.appType;
+        var fieldName = context.params.fieldName;
+
+        if (appType == 'mobileapp') {
+            var queryParams = request.getAllParameters();
+            var mobileValidatorModule = require('/extensions/mobileapp/validations.js');
+            var mobileValidator = new mobileValidatorModule.Validator(null);
+            var result = mobileValidator.validateFormField(fieldName, queryParams[fieldName],
+                                                           queryParams);
+            if (result.status != mobileValidatorModule.ValidationStatus.VALID) {
+                response.status = 422; // Unprocessable Entity (WebDAV; RFC 4918)
+                response.contentType = "application/json";
+                var content = {status: "Validation Error", message: result.message};
+                response.content = stringify(content);
+                return;
+            }
+            response.status = 200;
+            response.content = stringify({status: "Success"});
+        }
+    });
 
-    }
-    catch (e) {
-        log.debug('An asset of type: ' + shortName + ' could not be created.The following exception was thrown: ' + e);
-        response.sendError(500, 'An asset of type: ' + shortName + ' could not be created.Please check the server logs.');
-    }
+    routeManager.register('GET', 'publisher', '/publisher/api/validate/{appType}/form/name/{formName}', function (context) {
+
+        var appType = context.params.appType;
+        var formName = context.params.formName;
+
+        if (appType == 'mobileapp') {
+            var queryParams = request.getAllParameters();
+            var partialValidation = false;
+            if (queryParams['partial']) {
+                partialValidation = Boolean(queryParams['partial']);
+            }
+            var mobileValidatorModule = require('/extensions/mobileapp/validations.js');
+            var mobileValidator = new mobileValidatorModule.Validator(null);
+            var result = mobileValidator.validateForm(formName, queryParams, partialValidation);
+
+            if (result.status != mobileValidatorModule.ValidationStatus.VALID) {
+                response.status = 422; // Unprocessable Entity (WebDAV; RFC 4918)
+                response.contentType = "application/json";
+                var content = {
+                    status: "Validation Error",
+                    validationResults: result.validationResults
+                };
+                response.content = stringify(content);
+                return;
+            }
+            response.status = 200;
+            response.content = stringify({status: "Success"});
+        }
+    });
```

Comment:
L24 to L42 and L47 to L72 codes are look like same. Cann't we reuse the code by using one function?